client/src/App.tsx

```diff
@@ -1,30 +1,56 @@
-import { useEffect, useRef } from "react";
+import { useEffect, useLayoutEffect, useState } from "react";
 import * as THREE from "three";
+import { OrbitControls } from "three/examples/jsm/controls/OrbitControls";
+import Scene from "./Scene";
 
 function App() {
-  const ref = useRef<any>();
+  const [renderer, setRenderer] = useState<THREE.WebGLRenderer>();
+  const [camera, setCamera] = useState<THREE.PerspectiveCamera>();
 
   useEffect(() => {
-    var scene = new THREE.Scene();
-    var camera = new THREE.PerspectiveCamera(75, window.innerWidth / window.innerHeight, 0.1, 1000);
-    var renderer = new THREE.WebGLRenderer();
+    // WebGLRenderer
+    const renderer = new THREE.WebGLRenderer({
+      antialias: true,
+    });
+    renderer.shadowMap.enabled = true;
+    renderer.shadowMap.type = THREE.PCFSoftShadowMap;
+    renderer.setPixelRatio(window.devicePixelRatio);
     renderer.setSize(window.innerWidth, window.innerHeight);
-    ref.current?.appendChild(renderer.domElement);
-    var geometry = new THREE.BoxGeometry(1, 1, 1);
-    var material = new THREE.MeshBasicMaterial({ color: 0x00ff00 });
-    var cube = new THREE.Mesh(geometry, material);
-    scene.add(cube);
-    camera.position.z = 5;
-    var animate = function () {
-      requestAnimationFrame(animate);
-      cube.rotation.x += 0.01;
-      cube.rotation.y += 0.01;
-      renderer.render(scene, camera);
-    };
-    animate();
+    setRenderer(renderer);
+
+    // Camera
+    const fov = 60;
+    const aspect = window.innerWidth / window.innerHeight;
+    const near = 1.0;
+    const far = 1000;
+    const camera = new THREE.PerspectiveCamera(fov, aspect, near, far);
+    camera.position.set(0, 2, 5);
+    setCamera(camera);
+
+    // Controls
+    const controls = new OrbitControls(camera, renderer.domElement);
+    controls.target.set(0, 0, 0);
+    controls.update();
   }, []);
 
-  return <div ref={ref} />;
+  useLayoutEffect(() => {
+    function OnWindowResize() {
+      camera!.aspect = window.innerWidth / window.innerHeight;
```

Comment:
You could add `if (camera && rendered) { camera.aspect = ...  }` and so on to avoid using the `!` operator

Reply:
I mean I COULD but would I?

Reply:
I would.